### ISSUE TITLE
[dtest][consistent-topology-changes] topology coordinator retries rebuild forever instead of failing

### ISSUE THREAD

**Issue by temichus** [scylladb/scylladb#16740]
dtest test cases:
```
repair_based_node_operations_test.TestRepairBasedNodeOperations.test_disable_rbno
repair_based_node_operations_test.TestRepairBasedNodeOperations.test_disable_rbno_for_all_operations
```
failed with raft topology feature by timeout:
```
Failed: Timeout >2700.0s
```
due to the topology coordinator retrying to rebuild forever instead of failing.
it does try
```
INFO  2024-01-10 15:49:13,609 [shard 0:strm] storage_service - raft topology: rebuild from dc: (any dc)
INFO  2024-01-10 15:49:13,609 [shard 0:strm] storage_service - raft topology: retry streaming after previous attempt failed with streaming::stream_exception (Stream failed)
INFO  2024-01-10 15:49:13,609 [shard 0:strm] storage_service - enable_repair_based_node_ops=false, allowed_repair_based_node_ops={rebuild ,decommission ,removenode ,replace ,bootstrap}
INFO  2024-01-10 15:49:13,651 [shard 0:strm] range_streamer - Rebuild starts, nr_ranges_remaining=2798
```
and starts another rebuild every time

*Installation details*
Scylla version (or git commit hash): Scylla version 5.5.0~dev-0.20240109.f4f724921cc7

logs :
https://jenkins.scylladb.com/job/scylla-staging/job/artsiom_mishuta/job/dtest-pytest-gating/3/artifact/logs-full.release.000/

test_disable_rbno - WORKER_ID gw1
test_disable_rbno_for_all_operations - WORKER_ID gw3

**Comment by temichus** [comment]
REMINDER: remove issues related test disabling from https://github.com/scylladb/scylla-dtest/pull/3844 when issue closed

**Comment by gleb-cloudius** [comment]
Fixed by 51c40034f5465a3c469578a9b1d26c496f785816.
